modules/PatternUtils.js

```diff
@@ -167,23 +167,60 @@ export function formatPattern(pattern, params) {
       if (paramValue != null)
         pathname += encodeURI(paramValue)
     } else if (token === '(') {
+      parenHistory[parenCount] = ''
       parenCount += 1
     } else if (token === ')') {
+      const parenText = parenHistory.pop()
       parenCount -= 1
+
+      if (parenCount)
+        parenHistory[parenCount - 1] += parenText
+      else
+        pathname += parenText
     } else if (token.charAt(0) === ':') {
       paramName = token.substring(1)
       paramValue = params[paramName]
 
       invariant(
         paramValue != null || parenCount > 0,
-        'Missing "%s" parameter for path "%s"',
+        'missing "%s" parameter for path "%s"',
```

Comment:
Don't downcase this. It is going to be shown to a human as an error and should read as a proper sentence.